### PR TITLE
K8SPXC-656 Use PXC ImagePullPolicy for init containers

### PR DESCRIPTION
**Pull request by egegunes**
[K8SPXC-656](https://jira.percona.com/browse/K8SPXC-656) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Review by delgod — DISMISSED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| haproxy | passed |
| storage | passed |
| upgrade-haproxy | passed |
| scaling | passed |
| recreate | passed |
| init-deploy | passed |
| self-healing | passed |
| scaling-proxysql | passed |
| restore-to-encrypted-cluster | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| limits | failed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| pitr | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-consistency | passed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/848/commits/f4410353971cc6dea2cbb809ffb894966613d17c
image: `perconalab/percona-xtradb-cluster-operator:PR-848-f4410353`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| storage | passed |
| recreate | passed |
| haproxy | passed |
| big-data | passed |
| scaling | passed |
| self-healing | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| scaling-proxysql | passed |
| upgrade-proxysql | passed |
| security-context | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | passed |
| smart-update | passed |
| affinity | passed |
| operator-self-healing | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| pitr | passed |
| tls-issue-self | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-cert-manager | passed |
| upgrade-consistency | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/848/commits/69ccc0cc1e7620e319e306d7459867a82fc4ce0f
image: `perconalab/percona-xtradb-cluster-operator:PR-848-69ccc0cc`

File: e2e-tests/limits/compare/statefulset_no-limits-pxc-increased-oc.yml [hunk 1]
```diff
@@ -248,7 +248,7 @@ spec:
       initContainers:
         - command:
             - /pxc-init-entrypoint.sh
-          imagePullPolicy: Always
+          imagePullPolicy: IfNotPresent
```
Comment (qJkee) [inline e2e-tests/limits/compare/statefulset_no-limits-pxc-increased-oc.yml:251]: why do we change this value?

Comment (egegunes) [reply e2e-tests/limits/compare/statefulset_no-limits-pxc-increased-oc.yml:251]: With this PR, we set `ImagePullPolicy` to the same value for both init containers and regular ones. This change is to fix failing tests.

**Review by qJkee — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| upgrade-haproxy | passed |
| scaling-proxysql | passed |
| security-context | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/848/commits/00ec2fdd61b5d1b0166fc2b8145bccd6e9783cd6
image: `perconalab/percona-xtradb-cluster-operator:PR-848-00ec2fdd`

**Review by delgod — APPROVED** [review]

**Comment by blez** [comment]
These changes will make process of debug of `pxc-init-entrypoint.sh` difficult because we will have to rename operator image (which is used for init container) for every change.

**Comment by delgod** [comment]
@blez according to `CheckNSetDefaults` function, the default value is "Always".
so, this change shouldn't create the issues if user not specified value

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| scaling | passed |
| scaling-proxysql | passed |
| big-data | passed |
| security-context | passed |
| upgrade-haproxy | passed |
| haproxy | passed |
| recreate | passed |
| storage | passed |
| upgrade-proxysql | passed |
| init-deploy | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| limits | passed |
| self-healing-advanced | passed |
| monitoring-2-0 | passed |
| demand-backup | failed |
| affinity | passed |
| one-pod | passed |
| operator-self-healing | passed |
| auto-tuning | failed |
| proxysql-sidecar-res-limits | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/848/commits/00ec2fdd61b5d1b0166fc2b8145bccd6e9783cd6
image: `perconalab/percona-xtradb-cluster-operator:PR-848-00ec2fdd`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | passed |
| haproxy | passed |
| big-data | passed |
| storage | passed |
| scaling | passed |
| scaling-proxysql | passed |
| security-context | passed |
| recreate | passed |
| upgrade-proxysql | passed |
| self-healing | passed |
| smart-update | passed |
| upgrade-consistency | passed |
| init-deploy | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-advanced | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| operator-self-healing | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| pitr | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| demand-backup-encrypted-with-tls | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| scheduled-backup | failed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/848/commits/00ec2fdd61b5d1b0166fc2b8145bccd6e9783cd6
image: `perconalab/percona-xtradb-cluster-operator:PR-848-00ec2fdd`

**Review by blez — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| big-data | passed |
| scaling | passed |
| scaling-proxysql | passed |
| upgrade-haproxy | passed |
| security-context | passed |
| storage | passed |
| recreate | passed |
| self-healing | passed |
| restore-to-encrypted-cluster | passed |
| self-healing-advanced | passed |
| demand-backup | passed |
| operator-self-healing | passed |
| pitr | passed |
| demand-backup-encrypted-with-tls | passed |
| scheduled-backup | passed |
| haproxy | passed |
| init-deploy | passed |
| limits | passed |
| monitoring-2-0 | passed |
| affinity | passed |
| one-pod | passed |
| auto-tuning | passed |
| proxysql-sidecar-res-limits | passed |
| users | passed |
| tls-issue-self | passed |
| tls-issue-cert-manager | passed |
| tls-issue-cert-manager-ref | passed |
| validation-hook | passed |
| upgrade-proxysql | passed |
| smart-update | passed |
| upgrade-consistency | passed |

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/848/commits/00ec2fdd61b5d1b0166fc2b8145bccd6e9783cd6
image: `perconalab/percona-xtradb-cluster-operator:PR-848-00ec2fdd`

**Pull request merged**